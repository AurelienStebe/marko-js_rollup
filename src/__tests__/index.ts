import fs from "fs";
import path from "path";
import { rollup } from "rollup";
import css from "rollup-plugin-postcss";

const FIXTURES = path.join(__dirname, "fixtures");

fs.readdirSync(FIXTURES).forEach(fixture => {
  test(fixture, async () => {
    const dir = path.join(FIXTURES, fixture);
    const { default: config, snapshot } = await import(
      path.join(dir, "config.ts")
    );
    const bundle = await rollup({
      ...config,
      input: path.join(dir, config.input),
      plugins: [css({ extract: true }), ...config.plugins]
    });

    const generated = await bundle.generate({ dir, format: "cjs" });
    snapshot(
      (generated.output as any)
        .map(
          ({ fileName, source, code = source }) =>
            `// ${fileName}\n${code.toString("utf-8")}`
        )
        .join("\n")
    );
  });
});
