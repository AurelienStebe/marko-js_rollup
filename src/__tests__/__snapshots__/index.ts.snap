// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entry-no-hydrate 1`] = `
"// index.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var vdom = _interopDefault(require('marko/dist/vdom'));
var registryBrowser = _interopDefault(require('marko/dist/runtime/components/registry-browser'));
var renderer = _interopDefault(require('marko/dist/runtime/components/renderer'));
var defineComponent = _interopDefault(require('marko/dist/runtime/components/defineComponent'));
var vElement = _interopDefault(require('marko/dist/runtime/vdom/helpers/v-element'));
var _const = _interopDefault(require('marko/dist/runtime/vdom/helpers/const'));
var loadTag = _interopDefault(require('marko/dist/runtime/helpers/load-tag'));

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var classComponent = createCommonjsModule(function (module) {

var marko_template = module.exports = vdom.t(),
    marko_component = {
        onMount: function() {
          console.log(\\"mounted\\");
        }
      },
    marko_registerComponent = registryBrowser.r,
    marko_componentType = marko_registerComponent(\\"/@marko/rollup$latest/src/__tests__/fixtures/entry-no-hydrate/src/components/class-component.marko\\", function() {
      return module.exports;
    }),
    marko_const_nextId = _const(\\"4d364d\\"),
    marko_node0 = vElement(\\"div\\", {
        id: \\"class\\"
      }, \\"0\\", null, 0, 0, {
        i: marko_const_nextId()
      });

function render(input, out, __component, component, state) {

  out.n(marko_node0, component);
}

marko_template._ = renderer(render, {
    e_: marko_componentType
  }, marko_component);

marko_template.Component = defineComponent(marko_component, marko_template._);
});

var implicitComponent = createCommonjsModule(function (module) {

var marko_template = module.exports = vdom.t(),
    marko_registerComponent = registryBrowser.r,
    marko_componentType = marko_registerComponent(\\"/@marko/rollup$latest/src/__tests__/fixtures/entry-no-hydrate/src/components/implicit-component.marko\\", function() {
      return module.exports;
    }),
    class_component_tag = loadTag(classComponent),
    marko_attrs0 = {
        id: \\"implicit\\"
      };

function render(input, out, __component, component, state) {

  out.be(\\"div\\", marko_attrs0, \\"0\\", component);

  class_component_tag({}, out, __component, \\"1\\");

  out.ee();
}

marko_template._ = renderer(render, {
    d_: true,
    e_: marko_componentType
  });

marko_template.Component = defineComponent({}, marko_template._);
});

var src = createCommonjsModule(function (module) {

var marko_template = module.exports = vdom.t(),
    marko_registerComponent = registryBrowser.r,
    marko_componentType = marko_registerComponent(\\"/@marko/rollup$latest/src/__tests__/fixtures/entry-no-hydrate/src/index.marko\\", function() {
      return module.exports;
    }),
    implicit_component_tag = loadTag(implicitComponent),
    marko_attrs0 = {
        id: \\"page\\"
      };

function render(input, out, __component, component, state) {

  out.be(\\"div\\", marko_attrs0, \\"0\\", component);

  implicit_component_tag({}, out, __component, \\"1\\");

  out.ee();
}

marko_template._ = renderer(render, {
    d_: true,
    e_: marko_componentType
  });

marko_template.Component = defineComponent({}, marko_template._);
});

module.exports = src;

// index.css
div { color: green }"
`;

exports[`hydrate-entry 1`] = `
"// index.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var vdom = _interopDefault(require('marko/dist/vdom'));
var registryBrowser = _interopDefault(require('marko/dist/runtime/components/registry-browser'));
var renderer = _interopDefault(require('marko/dist/runtime/components/renderer'));
var defineComponent = _interopDefault(require('marko/dist/runtime/components/defineComponent'));
var vElement = _interopDefault(require('marko/dist/runtime/vdom/helpers/v-element'));
var _const = _interopDefault(require('marko/dist/runtime/vdom/helpers/const'));
var __rollup_marko_runtime__ = _interopDefault(require('marko/components'));

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var classComponent = createCommonjsModule(function (module) {

var marko_template = module.exports = vdom.t(),
    marko_component = {
        onMount: function() {
          console.log(\\"mounted\\");
        }
      },
    marko_registerComponent = registryBrowser.r,
    marko_componentType = marko_registerComponent(\\"/@marko/rollup$latest/src/__tests__/fixtures/hydrate-entry/src/components/class-component.marko\\", function() {
      return module.exports;
    }),
    marko_const_nextId = _const(\\"4d364d\\"),
    marko_node0 = vElement(\\"div\\", {
        id: \\"class\\"
      }, \\"0\\", null, 0, 0, {
        i: marko_const_nextId()
      });

function render(input, out, __component, component, state) {

  out.n(marko_node0, component);
}

marko_template._ = renderer(render, {
    e_: marko_componentType
  }, marko_component);

marko_template.Component = defineComponent(marko_component, marko_template._);
});

__rollup_marko_runtime__.init();

// index.css
div { color: green }"
`;

exports[`hydrate-entry-custom-init 1`] = `
"// index.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var vdom = _interopDefault(require('marko/dist/vdom'));
var registryBrowser = _interopDefault(require('marko/dist/runtime/components/registry-browser'));
var renderer = _interopDefault(require('marko/dist/runtime/components/renderer'));
var defineComponent = _interopDefault(require('marko/dist/runtime/components/defineComponent'));
var vElement = _interopDefault(require('marko/dist/runtime/vdom/helpers/v-element'));
var _const = _interopDefault(require('marko/dist/runtime/vdom/helpers/const'));
var __rollup_marko_runtime__ = _interopDefault(require('marko/components'));

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var classComponent = createCommonjsModule(function (module) {

var marko_template = module.exports = vdom.t(),
    marko_component = {
        onMount: function() {
          console.log(\\"mounted\\");
        }
      },
    marko_registerComponent = registryBrowser.r,
    marko_componentType = marko_registerComponent(\\"/@marko/rollup$latest/src/__tests__/fixtures/hydrate-entry-custom-init/src/components/class-component.marko\\", function() {
      return module.exports;
    }),
    marko_const_nextId = _const(\\"4d364d\\"),
    marko_node0 = vElement(\\"div\\", {
        id: \\"class\\"
      }, \\"0\\", null, 0, 0, {
        i: marko_const_nextId()
      });

function render(input, out, __component, component, state) {

  out.n(marko_node0, component);
}

marko_template._ = renderer(render, {
    e_: marko_componentType
  }, marko_component);

marko_template.Component = defineComponent(marko_component, marko_template._);
});

__rollup_marko_runtime__.init(\\"SOME_COMPONENTS\\");

// index.css
div { color: green }"
`;

exports[`hydrate-entry-skip-init 1`] = `
"// index.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var vdom = _interopDefault(require('marko/dist/vdom'));
var registryBrowser = _interopDefault(require('marko/dist/runtime/components/registry-browser'));
var renderer = _interopDefault(require('marko/dist/runtime/components/renderer'));
var defineComponent = _interopDefault(require('marko/dist/runtime/components/defineComponent'));
var vElement = _interopDefault(require('marko/dist/runtime/vdom/helpers/v-element'));
var _const = _interopDefault(require('marko/dist/runtime/vdom/helpers/const'));

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var classComponent = createCommonjsModule(function (module) {

var marko_template = module.exports = vdom.t(),
    marko_component = {
        onMount: function() {
          console.log(\\"mounted\\");
        }
      },
    marko_registerComponent = registryBrowser.r,
    marko_componentType = marko_registerComponent(\\"/@marko/rollup$latest/src/__tests__/fixtures/hydrate-entry-skip-init/src/components/class-component.marko\\", function() {
      return module.exports;
    }),
    marko_const_nextId = _const(\\"4d364d\\"),
    marko_node0 = vElement(\\"div\\", {
        id: \\"class\\"
      }, \\"0\\", null, 0, 0, {
        i: marko_const_nextId()
      });

function render(input, out, __component, component, state) {

  out.n(marko_node0, component);
}

marko_template._ = renderer(render, {
    e_: marko_componentType
  }, marko_component);

marko_template.Component = defineComponent(marko_component, marko_template._);
});

// index.css
div { color: green }"
`;

exports[`hydrate-split-component 1`] = `
"// index.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

require('marko/dist/runtime/components');
var __rollup_marko_runtime__ = _interopDefault(require('marko/components'));

var __rollup_marko_component__ = {
  onMount() {
    console.log(\\"mounted\\");
  }
};

__rollup_marko_runtime__.register(\\"/@marko/rollup$latest/src/__tests__/fixtures/hydrate-split-component/src/components/split-component.component-browser\\",__rollup_marko_component__);

__rollup_marko_runtime__.init();

// index.css
div { color: green }"
`;

exports[`manual-render 1`] = `
"// index.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var vdom = _interopDefault(require('marko/dist/vdom'));
var registryBrowser = _interopDefault(require('marko/dist/runtime/components/registry-browser'));
var renderer = _interopDefault(require('marko/dist/runtime/components/renderer'));
var defineComponent = _interopDefault(require('marko/dist/runtime/components/defineComponent'));
var vElement = _interopDefault(require('marko/dist/runtime/vdom/helpers/v-element'));
var _const = _interopDefault(require('marko/dist/runtime/vdom/helpers/const'));
var loadTag = _interopDefault(require('marko/dist/runtime/helpers/load-tag'));

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var classComponent = createCommonjsModule(function (module) {

var marko_template = module.exports = vdom.t(),
    marko_component = {
        onMount: function() {
          console.log(\\"mounted\\");
        }
      },
    marko_registerComponent = registryBrowser.r,
    marko_componentType = marko_registerComponent(\\"/@marko/rollup$latest/src/__tests__/fixtures/manual-render/src/components/class-component.marko\\", function() {
      return module.exports;
    }),
    marko_const_nextId = _const(\\"4d364d\\"),
    marko_node0 = vElement(\\"div\\", {
        id: \\"class\\"
      }, \\"0\\", null, 0, 0, {
        i: marko_const_nextId()
      });

function render(input, out, __component, component, state) {

  out.n(marko_node0, component);
}

marko_template._ = renderer(render, {
    e_: marko_componentType
  }, marko_component);

marko_template.Component = defineComponent(marko_component, marko_template._);
});

var implicitComponent = createCommonjsModule(function (module) {

var marko_template = module.exports = vdom.t(),
    marko_registerComponent = registryBrowser.r,
    marko_componentType = marko_registerComponent(\\"/@marko/rollup$latest/src/__tests__/fixtures/manual-render/src/components/implicit-component.marko\\", function() {
      return module.exports;
    }),
    class_component_tag = loadTag(classComponent),
    marko_attrs0 = {
        id: \\"implicit\\"
      };

function render(input, out, __component, component, state) {

  out.be(\\"div\\", marko_attrs0, \\"0\\", component);

  class_component_tag({}, out, __component, \\"1\\");

  out.ee();
}

marko_template._ = renderer(render, {
    d_: true,
    e_: marko_componentType
  });

marko_template.Component = defineComponent({}, marko_template._);
});

var template = createCommonjsModule(function (module) {

var marko_template = module.exports = vdom.t(),
    marko_registerComponent = registryBrowser.r,
    marko_componentType = marko_registerComponent(\\"/@marko/rollup$latest/src/__tests__/fixtures/manual-render/src/template.marko\\", function() {
      return module.exports;
    }),
    implicit_component_tag = loadTag(implicitComponent),
    marko_attrs0 = {
        id: \\"page\\"
      };

function render(input, out, __component, component, state) {

  out.be(\\"div\\", marko_attrs0, \\"0\\", component);

  implicit_component_tag({}, out, __component, \\"1\\");

  out.ee();
}

marko_template._ = renderer(render, {
    d_: true,
    e_: marko_componentType
  });

marko_template.Component = defineComponent({}, marko_template._);
});

template.render({}).appendTo(document.body);

// index.css
div { color: green }"
`;
